### PR TITLE
fix: do not run tests if the path to the directory doesn't exist

### PR DESCRIPTION
**Pull request by smnppKDAB**

### CODE REVIEW

File: src/core/knutcore.cpp [hunk 1]
```diff
@@ -98,6 +98,7 @@ void KnutCore::process(const QStringList &arguments)
         } else {
             spdlog::error("KnutCore::process - Root directory: {}, does not exist. Cannot open a new project!",
                           pathDir.absolutePath());
+            return;
```
Comment (LeonMatthesKDAB) [inline src/core/knutcore.cpp:101]: Hm, when testing this, Knut actually doesn't exit, it just hangs forever until I kill it, and the GUI also doesn't open.

So unless we open a GUI afterwards, we probably want to exit(1) instead?

**Review by LeonMatthesKDAB — CHANGES_REQUESTED** [review]

File: src/core/knutcore.cpp (hunk 1 above)
Comment (dfaure-kdab) [reply src/core/knutcore.cpp:101]: Heh I hadn't seen this comment yet ;)

[the remark this answers is not in the document]
Comment (LeonMatthesKDAB) [reply src/core/knutcore.cpp:101]: I think not even `qApp->exit()` will actually help here, as the event loop is not yet running?
At least that is suggested by: https://doc.qt.io/qt-6/qcoreapplication.html#exit

[the remark this answers is not in the document]
Comment (LeonMatthesKDAB) [reply src/core/knutcore.cpp:101]: But maybe it then picks it up in the `exec` call?
Unsure about that.

[the remark this answers is not in the document]
Comment (LeonMatthesKDAB) [reply src/core/knutcore.cpp:106]: At least for me locally this indeed does not work...

Knut is not exiting! @smnppKDAB Does it work for you?

**Review by LeonMatthesKDAB — CHANGES_REQUESTED** [review]
For me, Knut still isn't exiting when the project path doesn't exist.

File: src/core/knutcore.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (smnppKDAB) [reply src/core/knutcore.cpp]: now it's working for me

**Review by dfaure-kdab — APPROVED** [review]

**Review by LeonMatthesKDAB — APPROVED** [review]

**Pull request merged**